src/utils/RealEstateApi.ts

```diff
@@ -0,0 +1,50 @@
+class RealEstateApi {
+  readonly baseUrl: string;
+
+  readonly defaultHeaders: HeadersInit;
+
+  constructor(options: { baseUrl: string }) {
+    this.baseUrl = options.baseUrl;
+    this.defaultHeaders = {
+      'Content-Type': 'application/json; charset=utf-8',
+    };
+  }
+
+  private getHeaders(): HeadersInit {
+    const token = localStorage.getItem('token');
+
+    return {
+      ...this.defaultHeaders,
+      Authorization: `Bearer ${token}`,
+    };
+  }
+
+  private async get(url: string, options?: { headers?: HeadersInit }) {
+    return fetch(`${this.baseUrl}/${url}`, {
+      headers: options?.headers || this.defaultHeaders,
+      method: 'GET',
+    });
+  }
+
+  private async post(url: string, options: { headers?: HeadersInit; body?: string }) {
```

Comment:
В метод post обычно хочется отдать объект с какими то данными, но не string, а уже в реализации метода будет преобразование JSON строку

Попробуй использовать тип unknown и body в POST думаю должен быть обязательным, так что можно убрать оператор опциональности - "?"
```suggestion
  private async post(url: string, options: { headers?: HeadersInit; body: unknown }) {
```